components/MultiPanel/index.ts

```diff
@@ -1,12 +1,183 @@
 import './styles.css';
 
+/**
+ * A multi-panel view that the user can add any number of 'panels'. 
+ * 'a panel' consists of two elements. even index element begomes heading, 
+ * and odd index element becomes the expandable content.
+ */
 export default class MultiPanel extends HTMLElement {
+
   constructor () {
     super();
+
+    // Watch for children changes.
+    new MutationObserver(() => this._childrenChange())
+      .observe(this, { childList: true });
   }
 
   connectedCallback () {
+    this._childrenChange();
+
+    const children : Element[] = Array.from(this.children);
+    for (let i : number = 0; i < children.length; i++){
+      const child : Element = children[i]
```

Comment:
Between that or `forEach((child, index) => { … })`, I don’t really have a preference. I do think either of those are nicer than the hand-written `for` loop though

Reply:
No preference here either.

Reply:
Why use an Array at all?

```js
let child = this.firstElementChild;
while (child) {
  // stuff
  child = child.nextElementSibling;
}
```

- nice balance of readability and performance
- avoids the array casting entirely

Reply:
Array operations are heavily optimized in V8 for actual arrays. The one-off cost of casting to an array is usually worth the later performance benefits when operating on the array (as opposed to an array-like object). (shameless https://www.youtube.com/watch?v=m9cTaYI95Zc plug in case anyone wants more info)

TL;DR Don’t think of the `Array.from()` as overhead.